source/MaterialXRuntime/RtCoreIo.cpp

```diff
@@ -487,8 +488,10 @@ RtApiType RtCoreIo::getApiType() const
 void RtCoreIo::read(const DocumentPtr& doc, RtCoreIo::ReadFilter filter)
 {
     PrvStage* stage = data()->asA<PrvStage>();
+    stage->clearAttributes();
```

Comment:
The clear can be optional. It was never clearing currently. Then you can append to a library stage or a non-library stage. However you cannot transfer over the document's attributes since there is no "import" logic, which tags new elements. This can be looked at.